src/search/merge_and_shrink/merge_selector.h

```diff
@@ -1,22 +1,31 @@
 #ifndef MERGE_AND_SHRINK_MERGE_SELECTOR_H
 #define MERGE_AND_SHRINK_MERGE_SELECTOR_H
 
+#include "../utils/logging.h"
+
 #include <string>
 #include <vector>
 
 class TaskProxy;
 
+namespace options {
+class Options;
+}
+
 namespace merge_and_shrink {
 class FactoredTransitionSystem;
+
 class MergeSelector {
 protected:
+    mutable utils::LogProxy log;
+
     virtual std::string name() const = 0;
-    virtual void dump_specific_options() const {}
+    virtual void dump_specific_options() const = 0;
```

Comment:
Why is this changed?

Reply:
I cannot comment on unchanged lines. In the concrete implementation of dump_specific_options(), why did you nto forward the LogProxy to the called dump functions? Even if you want to log their output always, if later the stream can be changed, we would overlook changing those calls.

Reply:
I'm not sure I understand your comment regarding "later changing this". I think the reason why I gave the ScoringFunction and MergeSelector classes their own log instances is to avoid using hard-coded "silent loggers" as it was previously done in MergeScoringFunctionMIASM. I thought it would be better if they could be configured separately instead. But I'm not convinced if that is indeed the best solution... And you are right again that this technically doesn't belong into this issue. Should I revert these two changed parts?

Reply:
My comment about unchanged lines was regarding the overridden version MergeSelectorScoreBasedFiltering::dump_specific_options.

You added a LogProxy to MergeSelectorScoreBasedFiltering, but then call another dump function without passing the logger in MergeSelectorScoreBasedFiltering::dump_specific_options.

Reply:
The issue is to provide the option to use local loggers. Your changes are more than required, thus, I would revert them.

Do you intend to do in the near future more changes to the logging in M&S (e.g. add the option in more place or use the log level in more outputs)? If yes, it would be easy to move your changes from here to that new issue.

Reply:
On the one hand, yes, but thinking about it a bit more, I think it is important that the two parts of the code which already use the verbosity option now switch to the new mechanism. Otherwise, we don't see how the new mechanism works. This is why I changed the search engines and merge-and-shrink to use the new mechanism.

Reply:
If we keep the code, then I suggest to use the local logger from MergeSelectorScoreBasedFiltering also in the `dump_options` calls written in 
```void MergeSelectorScoreBasedFiltering::dump_specific_options() const {
    for (const shared_ptr<MergeScoringFunction> &scoring_function
         : merge_scoring_functions) {
        scoring_function->dump_options();
    }
}```